### PR TITLE
Clarified package documentation for executor

### PR DESCRIPTION
**Pull request by cbednarski**
There was some confusion as to the purpose of executors in #232. I clarified the executor package documentation so the design is hopefully a bit more clear, including some examples.

### CODE REVIEW

File: client/executor/exec.go [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 // - Least privilege
 // - Resource constraints
 // - Process isolation
 //
-// A "platform" may be defined as coarsely as "Windows" or as specifically as
-// "linux 3.20 with systemd". This allows Nomad to use best-effort, best-
-// available capabilities of each platform to provide resource constraints,
-// process isolation, and security features, or otherwise take advantage of
-// features that are unique to that platform.
+// A "platform" may something like "windows" or "linux with systemd". Executors
```
Comment (jmitchell) [inline client/executor/exec.go]: s/may/may be/

Comment (cbednarski) [inline client/executor/exec.go]: :+1: Thanks!

**Comment by catsby** [comment]
:+1: more clear now

**Comment by dadgar** [comment]
Looks good to me

**Pull request merged**